addon-test-support/@ember/test-helpers/dom/fire-event.js

```diff
@@ -0,0 +1,227 @@
+import { merge } from '@ember/polyfills';
+
+const DEFAULT_EVENT_OPTIONS = { bubbles: true, cancelable: true };
+export const KEYBOARD_EVENT_TYPES = Object.freeze(['keydown', 'keypress', 'keyup']);
+const MOUSE_EVENT_TYPES = [
+  'click',
+  'mousedown',
+  'mouseup',
+  'dblclick',
+  'mouseenter',
+  'mouseleave',
+  'mousemove',
+  'mouseout',
+  'mouseover',
+];
+const FILE_SELECTION_EVENT_TYPES = ['change'];
+
+/**
+  @method fireEvent
+  @param {Element} element
+  @param {String} type
+  @param {Object} [options]
+  @returns {Event}
+
+  @private
+*/
+export default function fireEvent(element, type, options = {}) {
```

Comment:
why is this not using `nextTickPromise()`?

Reply:
Because it is private, not exported from the entry point, and only called from _within_ a nextTickPromise already. I can add some underscores too though 😜...